src/js/control-bar/play-toggle.js

```diff
@@ -83,20 +105,33 @@ class PlayToggle extends Button {
    */
   handlePause(event) {
     this.removeClass('vjs-playing');
-    this.addClass('vjs-paused');
+
+    if (!this.hasClass('vjs-paused')) {
+      this.addClass('vjs-paused');
+    }
     // change the button text to "Play"
     this.controlText('Play');
   }
 
   /**
    * Add the vjs-ended class to the element so it can change appearance
    *
+   * @param {EventTarget~Event} [event]
+   *        The event that caused this function to run.
+   *
+   * @listens Player#ended
    */
   handleEnded(event) {
     this.removeClass('vjs-playing');
-    this.addClass('vjs-ended');
+
+    if (!this.hasClass('vjs-ended')) {
```

Comment:
why not just keep this as unconditional?

Reply:
moved them back to unconditional as this is already a check in addClass.